src/core/agent/services/identifierService.ts

```diff
@@ -181,30 +272,45 @@ class IdentifierService extends AgentService {
       .addEndRole(identifier, "agent", this.props.signifyClient.agent!.pre);
     await addRoleOperation.op();
 
-    const isPending = !op.done;
-    if (isPending) {
-      op = await waitAndGetDoneOp(
-        this.props.signifyClient,
-        op,
-        2000 - (Date.now() - startTime)
-      );
-      if (!op.done) {
-        const pendingOperation = await this.operationPendingStorage.save({
-          id: op.name,
-          recordType: OperationPendingRecordType.Witness,
-        });
-        this.props.eventEmitter.emit<OperationAddedEvent>({
-          type: EventTypes.OperationAdded,
-          payload: { operation: pendingOperation },
-        });
-      }
-    }
+    const recordType = op.done
+      ? OperationPendingRecordType.Individual
+      : OperationPendingRecordType.Witness;
+
+    const pendingOperation = await this.operationPendingStorage.save({
+      id: op.name,
+      recordType,
+    });
+
+    this.props.eventEmitter.emit<OperationAddedEvent>({
+      type: EventTypes.OperationAdded,
+      payload: { operation: pendingOperation },
+    });
+
+    const isPending = true;
     await this.identifierStorage.createIdentifierMetadataRecord({
       id: identifier,
       ...metadata,
-      isPending: !op.done,
+      isPending,
     });
-    return { identifier, isPending: !op.done };
+
+    this.props.eventEmitter.emit<IdentifierAddedEvent>({
+      type: EventTypes.IdentifierAdded,
+      payload: { identifier: { id: identifier, ...metadata, isPending } },
+    });
+
+    const updatedRecord = await this.basicStorage.findById(
+      MiscRecordId.IDENTIFIERS_PENDING_CREATION
+    );
+
+    if (updatedRecord && Array.isArray(updatedRecord?.content?.name)) {
+      const isExist = updatedRecord.content.name.includes(name);
+      if (isExist) {
```

Comment:
Just call `shift` on the array instead of checking if it exists and then filtering. Less complex and more efficient.